### PR TITLE
feat: Add Vite configuration files and main.tsx

### PR DESCRIPTION
**Pull request by michaelto20**
This adds the necessary Vite configuration files and the main.tsx file to the AspireJavaScript.Vite sample project. The Vite configuration files include vite.config.ts, tsconfig.json and default.conf.template, which are essential for setting up the Vite development environment.  This commit sets up the basic structure of the Vite project.

### CODE REVIEW

**Review by IEvangelist — COMMENTED** [review]
This looks really good! Thank you, did you test the containerization aspect of this? Also, do you have a screen capture of the weather loading both locally and from the container?

**Comment by michaelto20** [comment]
@IEvangelist I did test it locally and in Azure and it worked so I assume the containerization aspect worked. Is there a different way to double check? I've just run it and here are the screen captures for both the dashboard and weather app when running locally and in Azure using `azd up`. Is this what you were looking for?
AspireDashboardAzure
AspireViteLocalDashboard
AspireViteWeatherLocal
ViteAspireWeatherAzure

**Comment by IEvangelist** [comment]
> @IEvangelist I did test it locally and in Azure and it worked so I assume the containerization aspect worked. Is there a different way to double check? I've just run it and here are the screen captures for both the dashboard and weather app when running locally and in Azure using `azd up`. Is this what you were looking for? AspireDashboardAzure AspireViteLocalDashboard AspireViteWeatherLocal ViteAspireWeatherAzure

Yeah, this is great! Thank you @michaelto20 - much appreciated.

File: samples/AspireWithJavaScript/AspireJavaScript.Vite/public/vite.svg
[the remark this answers is not in the document]
Comment (michaelto20) [reply samples/AspireWithJavaScript/AspireJavaScript.Vite/public/vite.svg:1]: Sorry just seeing this, what image are you referring to?

**Comment by michaelto20** [comment]
@IEvangelist I got distracted with work for a bit but trying to get this across the finish line. The build broke but I think it was related to an environmental error and not my code, as far as I can tell. Do you mind kicking off another build to see if it works? I tried to figure out how to do that but couldn't figure it out.

**Review by DamianEdwards — APPROVED** [review]
Thanks very much!

**Pull request merged**